modules/kafka/kafka_test.go

```diff
@@ -54,38 +75,150 @@ func TestKafka(t *testing.T) {
 	require.NoError(t, err)
 
 	_, _, err = producer.SendMessage(&sarama.ProducerMessage{
-		Topic: topic,
+		Topic: testTopic,
 		Key:   sarama.StringEncoder("key"),
-		Value: sarama.StringEncoder("value"),
+		Value: sarama.StringEncoder(testValue),
 	})
 	require.NoError(t, err)
 
 	<-done
 
 	require.Truef(t, strings.EqualFold(string(consumer.message.Key), "key"), "expected key to be %s, got %s", "key", string(consumer.message.Key))
-	require.Truef(t, strings.EqualFold(string(consumer.message.Value), "value"), "expected value to be %s, got %s", "value", string(consumer.message.Value))
+	require.Truef(t, strings.EqualFold(string(consumer.message.Value), testValue), "expected value to be %s, got %s", testValue, string(consumer.message.Value))
+
+	t.Run("BrokersByHostDockerInternal", func(t *testing.T) {
+		brokers, err := kafkaContainer.BrokersByHostDockerInternal(ctx)
+		require.NoError(t, err)
+
+		kcat, err := runKcatContainer(ctx, brokers, func(hc *container.HostConfig) {
+			hc.ExtraHosts = append(hc.ExtraHosts, "host.docker.internal:host-gateway")
+		}, nil)
+		testcontainers.CleanupContainer(t, kcat)
+		require.NoError(t, err)
+
+		l, err := kcat.Logs(ctx)
+		require.NoError(t, err)
+		defer l.Close()
+
+		assertKcatReadMsg(t, l)
+	})
+	t.Run("BrokersByContainerId", func(t *testing.T) {
+		brokers, err := kafkaContainer.BrokersByContainerId(ctx)
+		require.NoError(t, err)
+
+		kcat, err := runKcatContainer(ctx, brokers, nil, []string{net.Name})
+		testcontainers.CleanupContainer(t, kcat)
+		require.NoError(t, err)
+
+		l, err := kcat.Logs(ctx)
+		require.NoError(t, err)
+		defer l.Close()
+
+		assertKcatReadMsg(t, l)
+	})
+	t.Run("BrokersByContainerName", func(t *testing.T) {
+		brokers, err := kafkaContainer.BrokersByContainerName(ctx)
+		require.NoError(t, err)
+
+		kcat, err := runKcatContainer(ctx, brokers, nil, []string{net.Name})
+		testcontainers.CleanupContainer(t, kcat)
+		require.NoError(t, err)
+
+		l, err := kcat.Logs(ctx)
+		require.NoError(t, err)
+		defer l.Close()
+
+		assertKcatReadMsg(t, l)
+	})
```

Comment:
@stevenh Much happier with this after your suggestion to not use that giant closure here :saluting_face: